lib/src/accordion/Accordion.tsx

```diff
@@ -24,54 +32,100 @@ const DxcAccordion = ({
+                  }
+                  fontFamily={colorsTheme.accordion.titleLabelFontFamily}
+                  fontSize={colorsTheme.accordion.titleLabelFontSize}
+                  fontStyle={colorsTheme.accordion.titleLabelFontStyle}
+                  fontWeight={colorsTheme.accordion.titleLabelFontWeight}
+                  lineHeight="1.5em"
+                >
+                  {label}
+                </DxcTypography>
+              </AccordionLabel>
+              {assistiveText && (
+                <AccordionAssistiveText disabled={disabled}>
+                  <DxcTypography
+                    color={
+                      disabled
+                        ? colorsTheme.accordion.disabledAssistiveTextFontColor
+                        : colorsTheme.accordion.assistiveTextFontColor
+                    }
+                    fontFamily={colorsTheme.accordion.assistiveTextFontFamily}
+                    fontSize={colorsTheme.accordion.assistiveTextFontSize}
+                    fontStyle={colorsTheme.accordion.assistiveTextFontStyle}
+                    fontWeight={colorsTheme.accordion.assistiveTextFontWeight}
+                    letterSpacing={colorsTheme.accordion.assistiveTextLetterSpacing}
+                    lineHeight="1.5em"
+                  >
+                    {assistiveText}
+                  </DxcTypography>
+                </AccordionAssistiveText>
               )}
-              <AccordionLabel>{label}</AccordionLabel>
             </AccordionInfo>
-            {assistiveText && <AccordionAssistiveText disabled={disabled}>{assistiveText}</AccordionAssistiveText>}
-          </ExpansionPanelSummary>
-          <ExpansionPanelDetails>
-            <AccordionContent disabled={disabled}>
-              <BackgroundColorProvider color={colorsTheme.accordion.backgroundColor}>
-                {children}
-              </BackgroundColorProvider>
-            </AccordionContent>
-          </ExpansionPanelDetails>
-        </ExpansionPanel>
-      </DXCAccordion>
+            <CollapseIndicator disabled={disabled}>
+              {isExpanded ?? innerIsExpanded ? expandLess : expandMore}
+            </CollapseIndicator>
+          </AccordionTrigger>
+        </AccordionHeader>
+        {(isExpanded ?? innerIsExpanded) && (
+          <AccordionPanel
+            id={`accordion-panel-${id}`}
+            role="region"
+            aria-labelledBy={`accordion-${id}`}
+            padding={padding}
+          >
+            <BackgroundColorProvider color={colorsTheme.accordion.backgroundColor}>{children}</BackgroundColorProvider>
+          </AccordionPanel>
+        )}
+      </AccordionContainer>
     </ThemeProvider>
   );
 };
 
 const calculateWidth = (margin) => `calc(100% - ${getMargin(margin, "left")} - ${getMargin(margin, "right")})`;
 
-const DXCAccordion = styled.div`
+const AccordionContainer = styled.div<{ isExpanded: boolean; margin: Space | Margin; padding: Space | Padding }>`
```

Comment:
I think separating the types in a variable would be cleaner.

Reply:
Done 😉